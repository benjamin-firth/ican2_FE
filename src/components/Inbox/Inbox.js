import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { fetchData } from '../../utils/apiCalls';
import './Inbox.scss';
import MessagePreview from '../MessagePreview/MessagePreview';

const Inbox = () => {
  const [conversations, setConversations] = useState([]);
  const currentUser = useSelector(state => state.currentUser);

  const checkForMessages = () => {
    const body = {"query": "{conversations(userId: \""+ currentUser.id + "\") {senderId recipientId}}"};

    fetchData(body)
    .then(data => {
      setConversations(data.data.conversations)
    })
    .catch(error => error)
  }

  useEffect(() => checkForMessages(), [])

  const renderPreviews = () => {
    return conversations.map(conversation => {
      let otherMessengerId;
      if (conversation.senderId === currentUser.id) {
        otherMessengerId = conversation.recipientId;
      } else {
        otherMessengerId = conversation.senderId;
      }
      return <MessagePreview otherMessengerId={otherMessengerId}/>
    })
  }

  return (
    <section className='inbox'>
      <h3>you have chats open with:</h3>
      <div className='preview-container'>
        {renderPreviews()}
      </div>
    </section>
  );
}

export default Inbox;
