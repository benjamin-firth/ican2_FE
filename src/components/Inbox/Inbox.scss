@import '../../variables.scss';

.inbox {
  padding-top: 12vw;
  height: 100vw;
  background-color: $primarycolor;
  h3 {
    align-self: flex-start;
    color: $secondarycolor;
    font-family: $secondaryfont;
    font-size: 35px;
    margin: 4vw 0 0 2vw;
  }
}

.preview-container {
  display: flex;
  flex-wrap: wrap;
}
