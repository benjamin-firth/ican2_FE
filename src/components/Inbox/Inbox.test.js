import React from 'react';
import { shallow } from 'enzyme';
import Inbox from './Inbox';

describe('Inbox', () => {
  it('should match the snapshot', () => {
    expect(true).toBe(true);
  });

});
