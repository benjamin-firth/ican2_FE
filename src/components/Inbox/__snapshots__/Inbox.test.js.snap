// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inbox should match the snapshot 1`] = `
<section
  className="inbox"
>
  <MessagePreview />
  <MessagePreview />
</section>
`;
